doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+				          }
+                  description "User name.";
+                }                
+
+                leaf state {
+                    default "enabled";
+                    type string {
+                        pattern "enabled|disabled";
+                    }
+                }
+                
+                leaf role {
+                    default "admin";
+                    type string {
+                        pattern "admin|monitor";
+                    }
+                }                
+                
+                leaf full-name {
+                    type string;
+                }
+                
+                leaf hashed-password {
+                    mandatory true;
+                    type string;
+                }
+                
+                leaf password_history {
+                    type string;
+                  	description "List of hashed passwords seperated by comma."; 
+                }
+
+                leaf passwd_date_of_last_change {
+                    type uint32;
+                }                
+               
+            } /* end of list USER_TABLE_LIST */
+
+        } /* end of container USER_TABLE */
+
+    } /* end of container sonic-user-mgmt */
+
+} /* end of module sonic-user-mgmt */
+
+
+```
+
+```yang
+//filename:  sonic-role-mgmt.yang
+module sonic-role-mgmt {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-role-mgmt";
+	prefix role-mgmt;
+
+    description "Role Management YANG Module for SONiC OS";
+
+	revision 2022-12-01 {
+        description "First Revision";
+    }
+   container sonic-role-mgmt {
+
+        container ROLE_TABLE {
+
+            description "ROLE MANAGEMENT part of config_db.json";
+
+            list ROLE_TABLE_LIST {
+
+                key "name";
+
+                leaf name {
+                  type string {
+                        pattern '[a-zA-Z_]([a-zA-Z0-9_-]{0,31}|[a-zA-Z0-9_-]{0,30}$)';
+                        length 1..32 {
+                              error-message "Invalid length for the role-name.";
+                              error-app-tag rolename-invalid-length;
+                        }
+                  }
+                  description "Role name.";
+                }       
+
+                leaf primary_group {
+                    mandatory true;
+                    type string;
+                  	description "Primary linux group.";                    
+                }
+                
+                leaf secondary_groups {
+                    mandatory true;
+                    type string;
+                  	description "List of secondary linux groups seperated by comma.";                    
+                }                
+
+               
+            } /* end of list ROLE_TABLE_LIST */
+
+        } /* end of container ROLE_TABLE */
+
+    } /* end of container sonic-role-mgmt */
+
+} /* end of module sonic-role-mgmt */
+
+
+```
+
+##### Config CLI
```

Comment:
Hope we'll use auto Click command generator utility based on YANG model for these commands, please mention the commands that cant be auto-generated here.

Reply:
Sure